src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -845,13 +845,104 @@ public void remove_rollback_directory_if_exists(string packageName)
             return packageInstalls;
         }
 
-        /// <summary>
-        /// Sets the configuration for the package upgrade
-        /// </summary>
-        /// <param name="config">The configuration.</param>
-        /// <param name="packageInfo">The package information.</param>
-        /// <returns>The original unmodified configuration, so it can be reset after upgrade</returns>
-        private ChocolateyConfiguration set_package_config_for_upgrade(ChocolateyConfiguration config, ChocolateyPackageInformation packageInfo)
+		public ConcurrentDictionary<string, PackageResult> get_outdated(ChocolateyConfiguration config)
+		{
+			var packageManager = NugetCommon.GetPackageManager(
+				config,
+				_nugetLogger,
+				_packageDownloader,
+				installSuccessAction: null,
+				uninstallSuccessAction: null,
+				addUninstallHandler: false);
+
+			var outdatedPackages = new ConcurrentDictionary<string, PackageResult>();
+
+			set_package_names_if_all_is_specified(config, () => { config.IgnoreDependencies = true; });
+			var packageNames = config.PackageNames.Split(new[] { ApplicationParameters.PackageNamesSeparator }, StringSplitOptions.RemoveEmptyEntries).or_empty_list_if_null().ToList();
+
+			foreach (var packageName in packageNames)
+			{
+				var repository = packageManager.SourceRepository;
+
+				bool isRemote = IsRepositoryRemote(repository);
+
+				IPackage latestPackage;
+
+				if (isRemote)
+				{
+					latestPackage = repository.GetPackages().Where(x => x.Id.ToLower() == packageName && x.IsLatestVersion).SingleOrDefault();
+				}
+				else
+				{
+					latestPackage = repository.FindPackage(packageName);

```

Comment:
What happened to the method you had before? It was legit, I mentioned it was weird but then said nevermind.

Reply:
I changed it in https://github.com/chocolatey/choco/pull/1495/commits/4fa560fd785a028abf23a1f871cf6592b30e392d because LastOrDefault is not supported in remote repository and in this commit updated to use isRemote.
So if we query remote repository it gets only the latest package and when we call local repository we use the same method that old outdated used without additional parameters.

Reply:
I meant `repository.FindPackage()` for a repository of packages, which would also have multiple versions. Note the method I linked earlier shows you how to deal with this - https://github.com/chocolatey/choco/blob/cdf0938f43bf6f2ac52d7ec5d27e4746f0d16436/src/chocolatey/infrastructure.app/nuget/NugetList.cs#L130-L138

Reply:
Are you talking about supporting ```Prelease``` and ```IsListed``` like in this call ```packageManager.SourceRepository.FindPackage(packageName, version, config.Prerelease, allowUnlisted: false);``` 
Or are you talking about ```repository.FindPackage(packageName);``` not be able to get the latest version of package if there are multiple versions.